### PR TITLE
Feat/eps controller/integration

### PR DESCRIPTION
**Pull request by shqln**
## Intention

A description of the changes contained in the PR.

## Implementation

What decisions were made to implemenent the above

## Tests

What test cases are included

## Discussion

A place to ask questions of reviewers, clarify any uncertenties

### CODE REVIEW

**Review by NishqR — COMMENTED** [review]
unit tests working on my local

File: Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+void epsControllerComponent ::
+    SEND_DATA_CONOPS_cmdHandler(
+        const FwOpcodeType opCode,
+        const U32 cmdSeq,
+        Ref::InternalType c) {
+    // TODO
+    this->cmdResponse_out(opCode, cmdSeq, Fw::CmdResponse::OK);
+}
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp:107]: Based on discussions with @NishqR I don't think any of our components should have commands.

Commands should be carefully designed and added in later down the line, we want to reduce entry points to the system and carefully control where changes can be made

File: Ref/src/adapter/eps/controller/component/epsControllerComponent.fpp [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    # ----------------------------------------------------------------------
+
+    @ send commmand down to battery logic component
+    sync command SEND_BATTERY_COMMAND(c: InternalType)
+
+    @ request something from state
+    sync command GET_STATE(c: InternalType)
+
+    @ send data packet up to conops
+    sync command SEND_DATA_CONOPS(c: InternalType)
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/controller/component/epsControllerComponent.fpp:66]: These should also go

File: Ref/src/adapter/eps/controller/component/epsControllerComponent.hpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        const FwOpcodeType opCode, /*!< The opcode*/
+        const U32 cmdSeq,          /*!< The command sequence number*/
+        Ref::InternalType c);
+
+    //! Implementation for SEND_DATA_CONOPS command handler
+    //! send data packet up to conops
+    void SEND_DATA_CONOPS_cmdHandler(
+        const FwOpcodeType opCode, /*!< The opcode*/
+        const U32 cmdSeq,          /*!< The command sequence number*/
+        Ref::InternalType c);
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/controller/component/epsControllerComponent.hpp:115]: And these

File: Ref/src/adapter/eps/controller/component/test/ut/Tester.cpp [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+
+  // ----------------------------------------------------------------------
+  // Tests
+  // ----------------------------------------------------------------------
+
+  void Tester ::
+    toDo()
+  {
+    // TODO
+  }
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/controller/component/test/ut/Tester.cpp:53]: This code does nothing and should go

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    time get port timeGetOut
+
+    # ----------------------------------------------------------------------
+    # Commands
+    # ----------------------------------------------------------------------
+
+    @ Send battery level
+    async command SEND_BATTERY_LEVEL(
+                           battery: F32 @< the battery level to send through
+                         )
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp:48]: Also should go

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.cpp [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+// Handler implementations for user-defined typed input ports
+// ----------------------------------------------------------------------
+
+Ref::ReturnType epsBatteryComponent ::
+    epsBatLogicIn_handler(
+        const NATIVE_INT_TYPE portNum,
+        F32 battery_level) {
+    // TODO return
+    this->cmdResponse_out(portNum, battery_level);
+}
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.cpp:49]: go

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.hpp [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        //! Send battery level
+        void
+        SEND_BATTERY_LEVEL_cmdHandler(
+            const FwOpcodeType opCode, /*!< The opcode*/
+            const U32 cmdSeq,          /*!< The command sequence number*/
+            F32 battery                /*!<
+                           the battery level to send through
+                           */
+        );
+};
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.hpp:72]: this too

File: Ref/src/adapter/eps/hardware/payloadPower/component/epsPayloadPowerComponent.hpp [hunk 8]
```diff
@@ -13,79 +13,75 @@
...
+        void
+        PAYLOAD_POWER_CMD_cmdHandler(
+            const FwOpcodeType opCode, /*!< The opcode*/
+            const U32 cmdSeq,          /*!< The command sequence number*/
+            bool power                 /*!<
+                            boolean val, true -> ON, false -> OFF
+                            */
+        );
+};
+
```
Comment (daniel-digby) [inline Ref/src/adapter/eps/hardware/payloadPower/component/epsPayloadPowerComponent.hpp:84]: more rip

**Review by daniel-digby — CHANGES_REQUESTED** [review]
Few QA points:
- Rebase changes before opening PRs, normally a reviewer would ignore the pr if it still had conflicts
- In future ensure your commits messages are sensible, it was hard to follow changes when reviewing
- Fill out the pr description before merging please.

Have left requested changes around handlers, @NishqR can weigh in if there are any questions.

File: Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp (hunk 1 above)
Comment (NishqR) [reply Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp:107]: yeah no commands for now - at least not in these components, if we do ever end up having commands from ground we'll have to think of a more centralized way to handle them. 

But yeah we should probably get rid of these - I'm guilty of having them in some other branches too (just for exploring and testing functionality) 

Not sure if they can just be simply yeeted tho (as in lines deleted from all files) without throwing other errors

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp (hunk 5 above)
Comment (NishqR) [reply Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp:48]: old battery component, can ignore

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.cpp (hunk 6 above)
Comment (NishqR) [reply Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.cpp:49]: old battery component, can ignore

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.hpp (hunk 7 above)
Comment (NishqR) [reply Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.hpp:72]: old battery component, can ignore

File: Ref/src/adapter/eps/hardware/payloadPower/component/epsPayloadPowerComponent.hpp (hunk 8 above)
Comment (NishqR) [reply Ref/src/adapter/eps/hardware/payloadPower/component/epsPayloadPowerComponent.hpp:84]: old payload power component, dont know how that ended up here 🤷🏼‍♂️

File: Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp (hunk 1 above)
Comment (NishqR) [reply Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp:107]: nvm can confirm they can just be yeeted

Comment (NishqR) [reply Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp:107]: just need to make sure to remove em from .fpp, .cpp and .hpp files @shqln

Comment (NishqR) [reply Ref/src/adapter/eps/controller/component/epsControllerComponent.cpp:107]: yeah yeeting causing errors when I try to run the unit tests again - errors aren't exactly clear

File: Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp (hunk 5 above)
Comment (daniel-digby) [reply Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp:48]: if these are old the whole component should be removed from the codebase - we should not have dead code living outside of the commit log

Comment (daniel-digby) [reply Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp:48]: if we ever need to view the old code we can just look on github @NishqR

Comment (NishqR) [reply Ref/src/adapter/eps/hardware/batteryLevel/component/epsBatteryComponent.fpp:48]: ye good point, agreed

**Comment by daniel-digby** [comment]
closing, will re-implement when EPS slice is returned to
